src/middlewared/middlewared/plugins/disk_/wipe.py

```diff
@@ -67,59 +70,10 @@ async def wipe(self, job, dev, mode, sync, options):
         """
         Performs a wipe of a disk `dev`.
         It can be of the following modes:
-          - QUICK: clean the first few and last megabytes of every partition and disk
+          - QUICK: clean the first and last 32 megabytes on `dev`
           - FULL: write whole disk with zero's
           - FULL_RANDOM: write whole disk with random bytes
         """
         await self.middleware.call('disk.swaps_remove_disks', [dev], options)
-
-        if osc.IS_FREEBSD:
-            await self.middleware.call('disk.remove_disk_from_graid', dev)
-
-        # First do a quick wipe of every partition to clean things like zfs labels
-        if mode == 'QUICK':
-            for part in await self.middleware.call('disk.list_partitions', dev):
-                await self.wipe_quick(part['name'], part['size'])
-
-        await self.middleware.call('disk.destroy_partitions', dev)
-
-        if mode == 'QUICK':
-            await self.wipe_quick(dev)
-        else:
-            size = await self.middleware.call('disk.get_dev_size', dev) or 1
-
-            proc = await Popen([
-                'dd',
-                'if=/dev/{}'.format('zero' if mode == 'FULL' else 'random'),
-                f'of=/dev/{dev}',
-                'bs=1M',
-            ], stdout=subprocess.DEVNULL, stderr=subprocess.PIPE)
-
-            try:
-                async def dd_wait():
-                    while True:
-                        if proc.returncode is not None:
-                            break
-                        os.kill(proc.pid, signal.SIGUSR1 if osc.IS_LINUX else signal.SIGINFO)
-                        await asyncio.sleep(1)
-
-                asyncio.ensure_future(dd_wait())
-
-                while True:
-                    line = await proc.stderr.readline()
-                    if line == b'':
-                        break
-                    line = line.decode()
-                    reg = RE_DD.search(line)
-                    if reg:
-                        speed = float(reg.group(2)) if osc.IS_LINUX else int(reg.group(2))
-                        if osc.IS_LINUX:
-                            mapping = {'gb': 1024 * 1024 * 1024, 'mb': 1024 * 1024, 'kb': 1024, 'b': 1}
-                            speed = int(speed * mapping[reg.group(3).lower()])
-                        job.set_progress((int(reg.group(1)) / size) * 100, extra={'speed': speed})
-            except asyncio.CancelledError:
-                proc.kill()
-                raise
-
-        if sync:
-            await self.middleware.call('disk.sync', dev)
```

Comment:
The `13.0-stable` PR had:

`await self.middleware.call('disk.remove_disk_from_graid', dev)`

... at this location. I assume this was omitted on purpose ?

Reply:
Right because that only applies to freeBSD 😄

Reply:
Kewl... just making sure 👍🏼